tidymodel-prediction-workflows-inside-snowflake-with-orbital/tidymodel-prediction-workflows-inside-snowflake-with-orbital.md

```diff
@@ -0,0 +1,1145 @@
+```
+
+Notice that you can see all the pre-processing steps and model prediction steps.
+
+### Run predictions in Snowflake
+
+Now that we've converted our tidymodels workflow object to SQL with orbital, we can 
+run model predictions inside Snowflake. 
+
+Calling `predict()` with our orbital object will run the SQL code we saw above directly
+in Snowflake. 
+
+```r
+start_time <- Sys.time()
+preds <- 
+  predict(orbital_obj, lendingclub_dat) |>
+  compute(name = "LENDING_CLUB_PREDICTIONS_TEMP")
+
+end_time <- Sys.time()
+  
+preds
+```
+
+```
+# Source:   SQL [?? x 1]
+# Database: Snowflake 8.44.2[@Snowflake/LENDING_CLUB]
+   .pred
+   <dbl>
+ 1  13.6
+ 2  17.0
+ 3  12.5
+ 4  13.6
+ 5  11.9
+ 6  16.8
+ 7  12.3
+ 8  11.3
+ 9  14.0
+10  13.5
+# ℹ more rows
+# ℹ Use `print(n = ...)` to see more rows
+```
+
+> aside positive
+>
+> We've also used `dplyr::compute()` to force the query to compute—without `compute()`,
+> `predict()` would only be evaluated lazily. `compute()` saves the results to a temporary table. 
+> The `Sys.time()` calls will help us determine how fast orbital computed our predictions.
+
+To figure out how much orbital sped up our process, let's see how many predictions
+we computed:
+
+```r
+preds |> count()
+```
+
+```
+# Source:   SQL [1 x 1]
+# Database: Snowflake 8.44.2[@Snowflake/LENDING_CLUB]
+        n
+    <dbl>
+1 2260702
+```
+
+as well as how much time elapsed:
+
+```r
+end_time - start_time
+```
+
+```
+Time difference of 3.027164 secs
+```
+
+2,260,702 predictions in just 3.02 seconds—thanks to Snowflake and orbital!
+
+## Deploy model as a Snowflake view
+Duration: 5
+
+Next, we'll deploy our model so others can use it. We have a couple of options.
+
+One option is to write the predictions back to Snowflake as a permanent table by
+setting `temporary = FALSE` in `compute()`:
+
+```r
+preds <- 
+  predict(orbital_obj, lendingclub_dat) |>
+  compute(name = "LENDING_CLUB_PREDICTIONS")
+```
+
+Another is to write our model prediction function as a [view](https://docs.snowflake.com/en/user-guide/views-introduction). 
+
+> aside positive
+> 
+> A view allows query results to be accessed like a table. The query will execute
+> anytime the view is called. 
+
+
+To create this view, we first need to construct the SQL query that we want to store in the view.
+
+```r
+view_sql <-
+  lendingclub_dat |>
+  mutate(!!!orbital_inline(orbital_obj)) |> 
+  select(any_of(c("ID", ".pred"))) |> 
+  remote_query()
+```
+
+`orbital_inline()` allows us to use our orbital object in a dplyr pipeline. We can
+then use `select()` to select just the columns we need for our predictions view:
+`ID` (to identify the rows) and `.pred` (which contains the predictions).
+
+`dbplyr::remote_query()` converts our dplyr pipeline back into SQL. Let's take 
+a look at the result.
+
+```r
+view_sql
+```
+
+```
```

Comment:
this might end up a bit messy, can you set it to auto-fold and only expand when needed? It's a big block to scroll past.

Reply:
I think it will auto-fold when rendered on Snowflake's site. I'm having difficult getting their previewing to work right now, but I'll confirm before making the PR.